src/main/kotlin/io/csbroker/apiserver/service/post/PostServiceImpl.kt

```diff
@@ -0,0 +1,86 @@
+package io.csbroker.apiserver.service.post
+
+import io.csbroker.apiserver.common.enums.ErrorCode
+import io.csbroker.apiserver.common.exception.EntityNotFoundException
+import io.csbroker.apiserver.common.exception.UnAuthorizedException
+import io.csbroker.apiserver.controller.v1.post.response.CommentResponseDto
+import io.csbroker.apiserver.controller.v1.post.response.PostResponseDto
+import io.csbroker.apiserver.model.Post
+import io.csbroker.apiserver.model.PostLike
+import io.csbroker.apiserver.repository.post.CommentRepository
+import io.csbroker.apiserver.repository.post.PostLikeRepository
+import io.csbroker.apiserver.repository.post.PostRepository
+import io.csbroker.apiserver.repository.problem.ProblemRepository
+import io.csbroker.apiserver.repository.user.UserRepository
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+@Transactional(readOnly = true)
+class PostServiceImpl(
+    private val problemRepository: ProblemRepository,
+    private val postRepository: PostRepository,
+    private val postLikeRepository: PostLikeRepository,
+    private val userRepository: UserRepository,
+    private val commentRepository: CommentRepository,
+) : PostService {
+    override fun findByProblemId(problemId: Long, email: String?): List<PostResponseDto> {
+        val problem = problemRepository.findByIdOrNull(problemId) ?: throw EntityNotFoundException(
+            "${problemId}번 문제는 존재하지 않습니다",
+        )
+        val posts = postRepository.findAllByProblem(problem)
+        val comments = commentRepository.findAllByPostIn(posts)
+        val postLikes = postLikeRepository.findAllByPostIn(posts)
+        return posts.map {
+            PostResponseDto(
+                id = it.id!!,
+                content = it.content,
+                username = it.user.username,
+                likeCount = postLikes.count { postLike -> postLike.post == it }.toLong(),
```

Comment:
```suggestion
                likeCount = it.postLike.count()
```
`ProblemLike`처럼 `PostLike`를 가지고있으면 더 심플할 것 같은데요~!

Reply:
@ekzm8523 PostLike 테이블이 있긴한데, JPA 연관관계 매핑을 하도 안한지 오래돼서;;

일단 이거는 나중에 쿼리 개선으로 한 번에 처리해볼게용